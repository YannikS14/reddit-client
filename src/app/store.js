import { configureStore } from '@reduxjs/toolkit';
import postsReducer from '../features/posts/postsSlice';
import subredditReducer from '../features/subreddits/subredditsSlice';
import searchFieldReducer from '../features/searchField/searchFieldSlice';

export const store = configureStore({
  reducer: {
    posts: postsReducer,
    subreddits: subredditReducer,
    searchField: searchFieldReducer,
  },
});
