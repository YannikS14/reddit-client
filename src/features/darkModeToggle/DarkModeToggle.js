import React from 'react';
import { useDispatch } from 'react-redux';
import { updateTheme } from '../../appSlice';

export default function DarkModeToggle() {
  const dispatch = useDispatch();

  return (
    <div
      id="dark-mode-toggle"
      className="hidden sm:flex border-solid border-2 border-gray-300 dark:border-gray-500 rounded-md divide-x-2 divide-gray-300 dark:divide-gray-500 divide-solid"
    >
      <button
        id="light-mode"
        data-testid="btn-light-mode"
        className="px-4 py-2 text-primary dark:text-gray-50"
        onClick={() => dispatch(updateTheme('light'))}
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-6 w-6 pointer-events-none"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          />
        </svg>
      </button>
      <button
        id="dark-mode"
        data-testid="btn-dark-mode"
        className="px-4 py-2 dark:text-primary"
        onClick={() => dispatch(updateTheme('dark'))}
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-6 w-6 pointer-events-none"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
          />
        </svg>
      </button>
    </div>
  );
}
