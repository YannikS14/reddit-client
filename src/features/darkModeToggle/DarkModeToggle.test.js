import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import DarkModeToggle from './DarkModeToggle';

describe('DarkModeToggle component', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DarkModeToggle />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
