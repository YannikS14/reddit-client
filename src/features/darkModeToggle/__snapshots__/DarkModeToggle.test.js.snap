// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeToggle component matches the snapshot 1`] = `
<div
  className="hidden sm:flex border-solid border-2 border-gray-300 dark:border-gray-500 rounded-md divide-x-2 divide-gray-300 dark:divide-gray-500 divide-solid"
  id="dark-mode-toggle"
>
  <button
    className="px-4 py-2 text-primary dark:text-gray-50"
    id="light-mode"
    onClick={[Function]}
  >
    <svg
      className="h-6 w-6 pointer-events-none"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </button>
  <button
    className="px-4 py-2 dark:text-primary"
    id="dark-mode"
    onClick={[Function]}
  >
    <svg
      className="h-6 w-6 pointer-events-none"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
      />
    </svg>
  </button>
</div>
`;
